bolt-simple-taproot.md

```diff
@@ -0,0 +1,1141 @@
+argument in order to strengthen their nonce.
+
+#### Nonce Handling
+
+For commitment transactions, Due to the asymmetric state of the current
+revocation channels, two sets of public nonces need to be exchanged by each
+party: one for the local commitment, and one for the remote commitment.
+
+The nonce for the local commitment MUST be sent a priori, thus enabling the
+counterparty to create a valid partial signature.
+
+The nonce for the remote commitment _could_ be sent a priori, too, but for the
+purposes of simplicity and disambiguation, MUST be sent alongside the partial
+signature. To that end, we will create a new data type that will be reused in
+multiple TLV extensions described further down in this document.
+
+The exception to this is the co-op close flow: as there's only a single message
+to sign, we only require a single pair of nonces. These nonces can also be sent
+right as the co-op flow begins.
+
+#### Signing
+
+Once public nonces have been exchanged, the `NonceAgg` algorithm is used to
+combine the public nonces into the aggregate nonce which will be a part of the
+final signature.
+
+The `Sign` method is used to generate a valid partial signature, with the
+`PartialSigVerify` algorithm used to verify each signature.
+
+Once all partial signatures are known, the `PartialSigAgg` algorithm is used to
+aggregate the partial signature into a final, valid [BIP
+340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki) Schnorr
+signature.
+
+##### Partial Signature Encoding
+
+A standard `musig2` partial signature is simply the encoding of the `s` value,
+which is a 32-byte-element modulo the order of the group.
+
+Throughout this specification, we will be using a custom type,
+`PartialSignatureWithNonce`, comprised of the aforementioned `s` value, along
+with the 66-byte-encoding of the public nonce used to sign remote commitments:
+
+```
+s || point_1 || point_2
+```
+
+## Design Overview
+
+With the preliminaries out of the way, we provide a brief overview of the
+Simple Taproot Channel design.
+
+The multisig output becomes a single P2TR key, with `musig2` key aggregation
+used to combine two keys into one.
+
+For commitment transactions, we inherit the anchor output semantics, meaning
+there are two outputs used for CPFP, with all other scripts inheriting a `1
+CSV` restriction.
+
+The local output of the commitment transaction uses the revocation key as the
+taproot internal key, and commits to a single script path of the delay script.
+
+The remote output of the commitment transaction uses the `combined_funding_key`
+as the top-level internal key, and then commits to a normal remote script.
+
+Anchor outputs use the respective two parties' funding keys as the top-level
+internal key committing to the script `16 CSV`. If a local or remote output is
+missing from the commitment, its corresponding anchor must not be present,
+either.
```

Comment:
If they are HTLCs, even if no remote output, an anchor output must be present for the remote party according to today BOLT3: https://github.com/lightning/bolts/blob/master/03-transactions.md#to_local_anchor-and-to_remote_anchor-output-option_anchors (you have pending funds at stake!)

Reply:
Good point, clarified.